File: fix_quotes.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if (file.endsWith('.ts') || file.endsWith('.tsx')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk('./src');
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let newContent = content.replace(/\\'x-forwarded-for\\'/g, "'x-forwarded-for'");
    newContent = newContent.replace(/\\'127\.0\.0\.1\\'/g, "'127.0.0.1'");
    if (content !== newContent) {
        fs.writeFileSync(file, newContent, 'utf8');
        console.log('Fixed', file);
    }
});

File: fix_quotes.py
import os

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        code = f.read()
    
    code = code.replace(\"\\'x-forwarded-for\\'\", \"'x-forwarded-for'\")
    code = code.replace(\"\\'127.0.0.1\\'\", \"'127.0.0.1'\")
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(code)

for root, _, files in os.walk('src'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            fix_file(os.path.join(root, file))
